hw/ip/otp_ctrl/data/otp_ctrl.hjson

```diff
@@ -0,0 +1,473 @@
+          desc: '''This register holds information on the error that occurred.
+                To be checked after an error interrupt occurred.
+                TODO: add reference to error type here.
+                '''
+        }
+      ]
+    }
+    { name: "DIRECT_ACCESS_CMD",
+      desc: "OTP array command for direct access.",
+      swaccess: "r0w1c",
+      hwaccess: "hro",
+      hwext:    "true",
+      hwqe:     "true",
+      fields: [
+        { bits: "0",
+          name: "read",
+          desc: "Initiates a readout sequence that reads the location specified by !!DIRECT_ACCESS_ADDRESS and !!DIRECT_ACCESS_SIZE. The command places the read data in !!DIRECT_ACCESS_RDATA."
+        }
+        { bits: "1",
+          name: "write",
+          desc: "Initiates a programming sequence that writes the !!DIRECT_ACCESS_WDATA to the location specified by !!DIRECT_ACCESS_ADDRESS and !!DIRECT_ACCESS_SIZE."
+        }
+      ]
+    }
+    { name: "DIRECT_ACCESS_ADDRESS",
+      desc: "OTP array address for direct access.",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "9:0",
+          desc: '''This is the address for the OTP word to be read or written through
+          the direct access interface. Note that the address is 32bit aligned internally,
+          hence bit 1:0 are ignored.'''
+        }
+      ]
+    }
+    { name: "DIRECT_ACCESS_SIZE",
+      desc: "OTP array acces size for direct access.",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "1:0",
+          desc: '''This is the access size for for the OTP word to be read or written through
+          the direct access interface. Note that the address is always aligned to the
+          access size internally. 0 = 1 Byte, 1 = 2 Bytes, 2 = 4 Bytes, 3 = 8 Bytes.'''
+        }
+      ]
+    }
+    { multireg: { name:     "DIRECT_ACCESS_WDATA",
+                  desc:     '''Write data for direct access. Note that only 1s are effectively programmed into the array. 0s do not have any effect.
+                  ''',
+                  count:    "2", // 2 x 32bit = 64bit
+                  swaccess: "rw",
+                  hwaccess: "hro",
+                  hwqe:     "true",
+                  cname:    "WORD",
+                  fields: [
+                    { bits: "31:0"
+                    }
+                  ]
+                }
+    },
+    { multireg: { name:     "DIRECT_ACCESS_RDATA",
+                  desc:     '''Read data for direct access.
+                  ''',
+                  count:    "2", // 2 x 32bit = 64bit
+                  swaccess: "ro",
+                  hwaccess: "hwo",
+                  cname:    "WORD",
+                  fields: [
+                    { bits: "31:0"
+                      desc: ""
+                    }
+                  ]
+                }
+    },
+
+    ///////////////////////
+    // Integrity Digests //
+    ///////////////////////
+    { skipto: "0x050" }
+
```

Comment:
do you think these registers should actually be hrw? It seems like it could also be used to indicate to a user that this parition has already been locked (ie, during init, otp checks the status and reflects it here)

Reply:
so these are just the calculation triggers. once they have been calculated and burnt into OTP, you can read the full digest in the regs further below, and writing to these calculation triggers becomes a NOP.